org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/serviceevent/ServiceEventPredicates.java

```diff
@@ -0,0 +1,173 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *******************************************************************************/
+
+package org.osgi.test.assertj.serviceevent;
+
+import java.util.Dictionary;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.osgi.framework.Constants;
+import org.osgi.framework.ServiceEvent;
+import org.osgi.framework.ServiceReference;
+import org.osgi.test.common.dictionary.Dictionaries;
+
+/**
+ * The Class ServiceEventPredicates.
+ */
+public class ServiceEventPredicates {
+
+	private ServiceEventPredicates() {}
+	/**
+	 * Returns a predicate that tests if the object Service event.
+	 *
+	 * @return the predicate
+	 */
+	public static Predicate<Object> serviceEvent() {
+		return e -> e instanceof ServiceEvent;
+	}
+
+	/**
+	 * Returns a predicate that tests if the event-type matches the
+	 * eventTypeMask.
+	 *
+	 * @param eventTypeMask the event type mask
+	 * @return the predicate
+	 */
+	// ServiceEvents
+	public static Predicate<ServiceEvent> type(final int eventTypeMask) {
+		return e -> (e.getType() & eventTypeMask) != 0;
+	}
+
+
+	/**
+	 * Returns a predicate that tests if the Service of the event matches the
+	 * objectClass.
+	 *
+	 * @param objectClass the objectClass
+	 * @return the predicate
+	 */
+	public static Predicate<ServiceEvent> objectClass(final Class<?> objectClass) {
+
+		return e -> {
+			Object classes = e.getServiceReference()
+				.getProperty(Constants.OBJECTCLASS);
+
+			if (classes != null && classes instanceof String[]) {
+				return Stream.of((String[]) classes)
+					.filter(Objects::nonNull)
+					.anyMatch(objectClass.getName()::equals);
+			}
+			return false;
+		};
+
+	}
+
+	/**
+	 * Returns a predicate that tests if the object Contain the service
+	 * property.
+	 *
+	 * @param key the key
+	 * @param value the value
+	 * @return the predicate
+	 */
+	public static Predicate<ServiceEvent> containServiceProperty(final String key, Object value) {
+		return e -> containsServiceProperties(Dictionaries.dictionaryOf(key, value)).test(e);
+	}
+
+	/**
+	 * Returns a predicate that tests if the object Contains the service
+	 * properties.
+	 *
+	 * @param map the map
+	 * @return the predicate
+	 */
+	public static Predicate<ServiceEvent> containsServiceProperties(Map<String, ?> map) {
+		return e -> {
+			ServiceReference<?> sr = e.getServiceReference();
+			List<String> keys = Stream.of(sr.getPropertyKeys())
```

Comment:
Isn't this just `Arrays.asList(sr.getPropertyKeys())`?

Reply:
fixed